File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = {}
    
    for p in participant:
        answer[p] = answer.get(p, 0) + 1
        
    for c in completion:
        answer[c] -= 1
        
    for key, value in answer.items():
        if value > 0:
            return key

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> people = new HashMap<>();

        for (String name : participant) {
            people.put(name, people.getOrDefault(name, 0) + 1);
        }

        for (String name : completion) {
            people.put(name, people.get(name) - 1);
        }

        for (Map.Entry<String, Integer> entry : people.entrySet()) {
            if (entry.getValue() > 0) {
                return entry.getKey();
            }
        }

        return "";
    }
}